_posts/2018-11-27-pool-features.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Pool features
+author: jmmichal
+layout: post
+identifier: pool_features
+---
+
+# Introduction
+
+Since the very first release, PMDK pools had internal feature flags. They were
+mostly a hidden implementation detail. The more observant users might have
+noticed pool features listing in [pmempool info][pmempool-info.1] output,
+but that's about it.
+
+[Release 1.5][release-1.5] introduced a set of new feature flags.
+And since it's imperative that system administrators have the ability to manage
+enabled features, we've added functionality that helps with that:
+
+* `pmempool` tool [commands for managing features][pmempool-feature.1]
+* *libpmempool* [API for managing features][pmempool-feature.3]
+* CTL namespace for
+[setting a `SHUTDOWN_STATE` initial value](#enabling-feature-at-creation-time)
+
+# Taxonomy of pool features
+
+Any one of the toggleable features can be disabled or enabled for the given pool
+and the support for those features might vary depending on the library version.
+All features are divided into two groups:
+
+* compatible features:
+	* `CHECK_BAD_BLOCKS`
+* incompatible features:
+	* `SINGLEHDR`
+	* `CKSUM_2K`
+	* `SHUTDOWN_STATE`
+
+Resulting in the following alternatives:
+
+- if a feature is disabled, it's disabled (obviously)
+- if a feautre is enabled, and supprted by the library, it's enabled
+- if a feature is enabled, and NOT supported, there are two things that might happen:
+	- if it's a compatible feature, it's disabled
+	- if it's a incompatible feature, the pool can NOT be opened
+
+## CKSUM_2K dependant features
+
+The first 4K of a pool is reserved for a pool header. The pool header is
+checksummed to ensure its consistency. If `CKSUM_2K` feature is enabled, only
+first 2K of the pool header is checksummed otherwise a whole 4K. This feature
+gives the library more flexibility which is needed for features that require
+extra pool header capacity. Currently, that's only `SHUTDOWN_STATE` feature.
+
+Some features will depend on the extra flexibility provided by `CKSUM_2K` feature.
```

Comment:
Merge this with the above sentence:
`Some features will depend on that. Currently, that's only `SHUTDOWN_STATE` feature.`

Reply:
Done.